superset/models/core.py

```diff
@@ -313,6 +314,9 @@ def get_sqla_engine(
         if connect_args:
             params["connect_args"] = connect_args
 
+        if user_name and self.impersonate_user and str(sqlalchemy_url).startswith('presto'):
```

Comment:
See ^^. You should update `get_configuration_for_impersonation` to handle this. Note given that this isn't part of the `connection`  key the  `get_configuration_for_impersonation` needs be refactored to be `get_connection_args_for_impersonation`.